src/Bookings.js

```diff
@@ -1,21 +1,117 @@
 import Search from "./Search.js";
-// import SearchResults from './SearchResults.js';
-// import FakeBookings from '../data/fakeBookings.json';
+import SearchResults from "./SearchResults.js";
+// import FakeBookings from "./data/fakeBookings.json";
+import AddBooking from "./AddBooking.js";
+
+const titles = [
+  ["ID", "id"],
+  ["Title", "title"],
+  ["First Name", "firstName"],
+  ["Surname", "surname"],
+  ["E-mail", "email"],
+  ["Room ID", "roomId"],
+  ["Check-in Date", "checkInDate"],
+  ["Check-out Date", "checkOutDate"],
+  ["Night Stay", "nightStay"]
+];
+
+const fetchURLs = [
+  "https://cyf-react.glitch.me/",
+  "https://cyf-react.glitch.me/delayed",
+  "https://cyf-react.glitch.me/error"
+];
 
 export default class Bookings extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoading: true,
+      error: null,
+      fakeBookings: [],
+      searchVal: "",
+      searchResults: null
+    };
+  }
+
+  componentDidMount() {
+    fetch(fetchURLs[0])
+      // fetch(fetchURLs[1])
+      // fetch(fetchURLs[2])
+      .then(res => {
+        if (res.status >= 200 && res.status < 300) {
+          return res;
+        } else {
+          throw new Error("HTTP error");
+        }
+      })
+      .then(res => res.json())
+      .then(data => {
+        // console.log(data);
+        this.setState({
+          isLoading: false,
+          fakeBookings: data
+        });
+      })
+      .catch(err => {
+        this.setState({
+          isLoading: false,
+          error: err
+        });
+      });
+    console.log(this.state.err);
+  }
+
+  displayAllBookings = () => {
+    this.setState({ searchVal: " " });
+  };
+
   search = searchVal => {
-    console.info("TO DO!", searchVal);
+    this.setState({ searchVal: searchVal });
+    this.setState({
+      searchResults: this.state.fakeBookings.filter(booking => {
+        return booking.firstName === searchVal || booking.surname === searchVal;
+      })
+    });
+  };
+
+  addBooking = newBooking => {
+    this.setState({
+      fakeBookings: this.state.fakeBookings.concat(newBooking)
+    });
   };
 
   render() {
-    return (
-      <div className="App-content">
-        <div className="container">
-          <Search search={this.search} />
-          {/* <SearchResults results={FakeBookings} /> */}
+    if (this.state.isLoading) {
+      return <span>Loading... 👽</span>;
+    } else if (this.state.error) {
+      return <span>Something went wrong 😭</span>;
+    } else {
+      const formFields = titles.slice(1, -1);
+      return (
+        <div className="App-content">
+          <div className="container">
+            <Search
+              search={this.search}
+              displayAllBookings={this.displayAllBookings}
+            />
+            <SearchResults
+              results={
+                this.state.searchVal
+                  ? this.state.searchResults
+                  : this.state.fakeBookings
```

Comment:
If you changed a few things slightly it would make your component a bit simpler.

When we fetch the data, we store the full list of bookings. Then here we do the filtering based on `this.state.searchVal` (instead of in the `search` method). That means that if we change `this.state.searchVal` the bookings will be re-filtered.

So you `displayAllBookings` method just has to set `searchVal` to be an empty string and it will re-filter to show all results.